hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/FileSourceBuilder.java

```diff
@@ -107,13 +111,17 @@ public FileSourceBuilder charset(@Nonnull Charset charset) {
      * The default local parallelism for this processor is 2 (or 1 if just 1
      * CPU is available).
      *
+     * @param readFileFn the function which reads line and emits immediate items from it.
+     *                   Gets file {@code Path} as parameter and returns Stream of immediate items.
      * @param mapOutputFn the function which creates output object from each
      *                    line. Gets the filename and line as parameters
+     * @param <I> the type of immediate items returned from file reading
      * @param <T> the type of the items the source emits
      */
-    public <T> BatchSource<T> build(BiFunctionEx<String, String, ? extends T> mapOutputFn) {
+    public <I, T> BatchSource<T> build(@Nonnull FunctionEx<? super Path, ? extends Stream<I>> readFileFn,
```

Comment:
does it make sense that if you want to provide a mapping, you also have to implement reading the files?

Reply:
I've left it for consistency with other methods. I don't see any specific benefit except user having some generic enrichment functions (which can be used inside readFileFn, so that's not a super-important argument)